wal_e/blobstore/file/calling_format.py

```diff
@@ -0,0 +1,97 @@
+import os
+from datetime import datetime
+
+
+def remove_empty_dirs(path):
+    """ removes empty dirs under a given path """
+    for root, dirs, files in os.walk(path):
+        for d in dirs:
+            dir_path = os.path.join(root, d)
+            if not os.listdir(dir_path):
+                os.rmdir(dir_path)
+
+
+def ensure_dir_exists(path):
+    """ create a directory if required """
+    dir_path = os.path.dirname(path)
+    if not os.path.exists(dir_path):
+        os.makedirs(dir_path)
+
+
+def common_dir_path(args, sep='/'):
+    """ return the highest common directory given a list of files """
+    return os.path.commonprefix(args).rpartition(sep)[0]
+
+
+def epoch_to_iso8601(timestamp):
+    return datetime.utcfromtimestamp(timestamp).isoformat()
+
+
+class FileKey(object):
+    def __init__(self, bucket, name):
+        self.bucket = bucket
+        self.name = name
+        self.path = os.path.join("/", name.strip("/"))
+        if os.path.isfile(self.path):
+            stat = os.stat(self.path)
+            self.last_modified = epoch_to_iso8601(stat.st_mtime)
+            self.size = stat.st_size
+
+    def get_contents_as_string(self):
+        fp = open(self.path, 'rb')
+        contents = fp.read()
+        fp.close()
+        return contents
+
+    def set_contents_from_file(self, fp):
+        ensure_dir_exists(self.path)
+        f = open(self.path, 'wb')
+        while True:
+            piece = fp.read(1024)
```

Comment:
Good call. Definitely doesn't make any sense for a buffer smaller than typical OS block or page size, or to fail to take advantage of a pertinent standard-library utility.